eng/scripts/CodeCheck.ps1

```diff
@@ -0,0 +1,119 @@
+#requires -version 5
+<#
+.SYNOPSIS
+This script runs a quick check for common errors, such as checking that Visual Studio solutions are up to date or that generated code has been committed to source.
+#>
+param(
+    [switch]$ci
+)
+
+$ErrorActionPreference = 'Stop'
+Set-StrictMode -Version 1
+Import-Module -Scope Local -Force "$PSScriptRoot/common.psm1"
+
+$repoRoot = Resolve-Path "$PSScriptRoot/../.."
+
+[string[]] $errors = @()
+
+function LogError {
+    param(
+        [Parameter(Mandatory = $true, Position = 0)]
+        [string]$message,
+        [string]$FilePath
+    )
+    if ($env:TF_BUILD) {
+        $prefix = "##vso[task.logissue type=error"
+        if ($FilePath) {
+            $prefix = "${prefix};sourcepath=$FilePath"
+        }
+        Write-Host "${prefix}]${message}"
+    }
+    Write-Host -f Red "error: $message"
+    $script:errors += $message
+}
+
+try {
+    if ($ci) {
+        $env:DOTNET_ROOT = "$repoRoot\.dotnet"
+        $env:PATH = "$env:DOTNET_ROOT;$env:PATH"
+
+        & $PSScriptRoot\..\common\build.ps1 -ci -prepareMachine -build:$false -restore:$false
```

Comment:
Suggest updating `build.ps1` to include `-noBuild` and `-noRestore` as done elsewhere

Reply:
Those are not accepted parameters in common/build.ps1 (https://github.com/aspnet/AspNetCore-Tooling/blob/ca9de74f4ead1fc544fb73b00212db9c677b8a9b/eng/common/build.ps1#L3-L24). This repo differs from aspnet/AspNetCore because it builds with Arcade